app/static/src/app/router.ts

```diff
@@ -0,0 +1,20 @@
+import { createRouter, createWebHistory, RouteComponent } from "vue-router";
+import SessionsPage from "./components/sessions/SessionsPage.vue";
+
+export function getRouter(appComponent: RouteComponent) {
+    const sessionsRoute = "/sessions";
+
+    let routeBase = new URL(window.location.href).pathname;
+    // User may be refreshing sessions page, in which case we want to remove /sessions from the route base
+    if (routeBase.endsWith(sessionsRoute)) {
+        routeBase = routeBase.slice(0, routeBase.length - sessionsRoute.length);
```

Comment:
I think you should use `Navigation Guards` here, specifically  `beforeRouteEnter`

Reply:
Sorry, I'm not quite sure what you're suggesting I should do here. I want to make all routes we're defining here relative to a certain base url, which depends both on the app being loaded, and whether we've initially loaded it from the back end with the `/sessions` suffix (so that we don't get the default component when we do that). I'm not sure what we could do in `beforeRouteEnter` to achieve the same thing - can you explain please?

Reply:
Maybe I don't understand the requirements well enough but what Insinuated was the below code where you could tell where a route is from
```
router.beforeEach((to, from, next) => {
  if (from.path == 'whatever' ) { //do something}
  else next()
})
```

Reply:
Yeah, I don't think that would quite help us here, as the important thing is the original url which is our starting point which other routes are then defined relative to. We might be able to fudge the routing but I think the history would end up looking  a bit strange. 

But you're right, this is a brittle approach. I might see if I can infer the base app url instead, either constructing it from info we already have from the back end, or using a regex.